src/CodeGen_LLVM.cpp

```diff
@@ -2956,7 +2944,26 @@ void CodeGen_LLVM::visit(const Call *op) {
                 }
             }
         }
-
+    } else if (op->is_intrinsic(Call::saturating_add) || op->is_intrinsic(Call::saturating_sub)) {
```

Comment:
This looks like code that will break when we add, e.g. 48-bit integers.

Reply:
LLVM does claim these support any bit width, and even use 4-bit integers as the example: https://llvm.org/docs/LangRef.html#llvm-sadd-sat-intrinsics

I agree it will likely break on a real target with 48-bit integers, but I think the best way to handle that would be to handle saturating_add/sub in that CodeGen class.